P5/Source/Specs/constraintDecl.xml

```diff
@@ -0,0 +1,96 @@
+        <sch:let name="schemes" value="//tei:constraintDecl[ not(ancestor::*[local-name(.) eq 'egXML'] ) ]/@scheme"/>
+        <sch:let name="disctinct_schemes" value="distinct-values( $schemes )"/>
+        <sch:assert test="count( $schemes ) eq count( $disctinct_schemes )">
+          Each &lt;constraintDecl> element should have a @scheme attribute that is distinct from that of all the other &lt;constraintDecl>s.
+        </sch:assert>
+      </sch:rule>
+    </constraint>
+  </constraintSpec>
+  <attList>
+    <attDef ident="scheme" usage="req">
+      <desc versionDate="2023-03-09" xml:lang="en">supplies the name of the language to which the declarations herein apply</desc>
+      <datatype><dataRef key="teidata.enumerated"/></datatype>
+      <valList type="semi">
+        <valItem ident="schematron">
+          <gloss versionDate="2016-09-27" xml:lang="en">ISO Schematron</gloss>
+        </valItem>
+      </valList>
+      <remarks versionDate="2023-03-09" xml:lang="en">
+        <p>The declarations contained in a particular
+        <gi>constraintDecl</gi> (including <att>queryBinding</att>, if
+        present) apply to the <gi>constraintSpec</gi> elements whose
+        <att>scheme</att> matches the <att>scheme</att> of the
+        <gi>constraintDecl</gi>.</p>
+      </remarks>
+    </attDef>
+    <attDef ident="queryBinding" usage="rec">
+      <gloss xml:lang="en" versionDate="2023-03-09">query language binding</gloss>
+      <desc xml:lang="en" versionDate="2023-03-09">specifies the query
+      language binding for rule-based schema expressions in
+      <gi>constraintSpec</gi> elements that have a matching
+      <att>scheme</att> attribute</desc>
+      <datatype><dataRef key="teidata.enumerated"/></datatype>
+      <valList type="semi">
+        <valItem ident="exslt"/>
+        <valItem ident="stx"/>
+        <valItem ident="xslt"/>
+        <valItem ident="xslt2"/>
+        <valItem ident="xslt3"/>
+        <valItem ident="xpath"/>
+        <valItem ident="xpath2"/>
+        <valItem ident="xpath3"/>
+        <valItem ident="xpath31"/>
+        <valItem ident="xquery"/>
+        <valItem ident="xquery3"/>
+        <valItem ident="xquery31"/>
+      </valList>
+      <remarks versionDate="2023-03-09" xml:lang="en">
+        <p>The suggested values above are the values reserved by the
+        Schematron specification. Only <val>exslt</val>,
+        <val>stx</val>, <val>xslt</val>, <val>xslt2</val>,
+        <val>xslt3</val>, <val>xpath2</val>, and <val>xpath3</val> are
+        defined by the specification. Most processors only support one
+        or more of <val>xslt</val>, <val>xslt2</val>, and
+        <val>xslt3</val>.</p>
+      </remarks>
+    </attDef>
+  </attList>
+  <exemplum xml:lang="en">
+    <egXML xmlns="http://www.tei-c.org/ns/Examples">
+      <constraintDecl scheme="schematron" queryBinding="xslt3">
+        <sch:ns prefix="wwp" uri="http://www.wwp.northeastern.edu/ns/textbase"/>
```

Comment:
This code does not function to deliver a specialized namespace to the constraintSpecs that evoke it, but we don't see this until we attempt to validate XML with the output RNG. I had to declare the `sch:ns` later on within the `<constraintSpec>` (immediately within a `<constraint>` element) for it to be recognized by Schematron and function for validation.